src/network/macvlan.rs

```diff
@@ -165,7 +166,48 @@ fn setup(
     opts.info_data = Some(InfoData::MacVlan(vec![InfoMacVlan::Mode(
         data.macvlan_mode,
     )]));
-    host.create_link(opts).wrap("create macvlan interface")?;
+    let mut result = host.create_link(opts.clone());
+    // Sigh, the kernel creates the interface first in the hostns before moving it into the netns.
+    // Therefore it can fail with EEXIST if the name is already used on the host. Create the link
+    // with tmp name, then rename it in the netns.
+    // If you change the iterations here make sure to match the number in early return case below as well.
+    for i in 0..3 {
```

Comment:
Why is this iteration needed ?

Reply:
there is no guarantee that the random name is free, I think trying three times is enough.

Reply:
but then why below has a check to return error when iterator is on second counter, let me mark that line

Reply:
This one https://github.com/containers/netavark/pull/470#discussion_r1006009190

Reply:
ah nvm I think you want to throw error on the last try.

Reply:
no, for i in 0..3 means 0,1,2. like any classic for loop, for(int = 0; i<3;i++){}
therefore we need to return the error when we are in the last iteration.